test/Microsoft.Azure.WebJobs.Extensions.Kafka.UnitTests/DiagnosticTest.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Extensions.Kafka.Diagnostics;
+using System.Collections;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka.UnitTests
+{
+    public class DiagnosticTest
+    {
+        [Fact]
+        public void SingleActivityProvider_Should_Create_Activity()
+        {
+            var consumerGroup = "my-consumer-group";
+            var numActivityTags = 7;
+            var traceId = ActivityTraceId.CreateRandom();
+            var spanId = ActivitySpanId.CreateRandom();
+            string traceparent = "00-" + traceId + "-" + spanId + "-" + "01";
+            var kafkaEvent = new KafkaEventData<string, string>("key", "value");
+            kafkaEvent.Headers.Add("traceparent", Encoding.ASCII.GetBytes(traceparent));
+            kafkaEvent.Topic = "mytopic";
+            kafkaEvent.Partition = 1;
+
+            var activityListener = new ActivityListener
+            {
+                ShouldListenTo = _ => true,
+                Sample = (ref ActivityCreationOptions<ActivityContext> _) => ActivitySamplingResult.AllData,
+            };
+            ActivitySource.AddActivityListener(activityListener);
+            var singleEventActivityProvider = new SingleEventActivityProvider(kafkaEvent, consumerGroup);
```

Comment:
Can we try this with DI framework ??
Same way we are doing for KafkaProducer binding

Reply:
My bad deleting this comment